ovh/resource_ovh_cloud_kubernetes_cluster.go

```diff
@@ -0,0 +1,351 @@
+				err := resourceCloudKubernetesClusterRead(d, meta)
+				return []*schema.ResourceData{d}, err
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OVH_PROJECT_ID", nil),
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"control_plane_is_up_to_date": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"is_up_to_date": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"next_upgrade_versions": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"nodes_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Default:  "GRA5",
+				Optional: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"update_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"client_certificate": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"client_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"cluster_ca_certificate": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"kubeconfig": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+type PublicCloudKubernetesClusterCreateOpts struct {
+	Name    string `json:"name"`
+	Region  string `json:"region"`
+	Version string `json:"version"`
+}
+
+func resourceCloudKubernetesClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	projectId := d.Get("project_id").(string)
+	params := &PublicCloudKubernetesClusterCreateOpts{
+		Name: d.Get("name").(string),
+		// TODO: Write check to ensure selected region is available for Kubernetes cluster
+		Region: d.Get("region").(string),
+		// TODO: Write check to ensure version is supported
+		Version: d.Get("version").(string),
+	}
+
+	r := &PublicCloudKubernetesClusterResponse{}
+
+	log.Printf("[DEBUG] Will create public cloud kubernetes cluster: %s", params)
+
+	d.Partial(true)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/kube", projectId)
+
+	err := config.OVHClient.Post(endpoint, params, r)
+	if err != nil {
+		return fmt.Errorf("calling Post %s with params %s:\n\t %q", endpoint, params, err)
+	}
+
+	// This is a fix for a weird bug where the cluster is not immediately available on API
+	bugFixWait := &resource.StateChangeConf{
+		Pending:    []string{"NOT_FOUND"},
```

Comment:
I suggest to have a string enum for statuses (maybe with [stringer](https://pkg.go.dev/golang.org/x/tools/cmd/stringer)?).
This will lead to a easy to read overview of all statuses.
WDYT?

Reply:
I don't know how to implement it, can you provide an example?

Reply:
I changed my mind about string, it would be a problem for *unknown statuses* (in case OVHcloud adds more statuses later).
So without stringer it could be implemented that way:
```go
type KubernetesClusterStatus string

// The list of all statuses is here: https://api.ovh.com/console/#/cloud/project/%7BserviceName%7D/kube/%7BkubeId%7D#GET
const (
    Ready KubernetesClusterStatus = "READY"
    Installing KubernetesClusterStatus = "INSTALLING"
    Updating KubernetesClusterStatus = "UPDATING"

    // This status is for cluster not yet available on API
    NotFound = "NOT_FOUND"
)

type PublicCloudKubernetesClusterResponse struct {
	Id                     string   `json:"id"`
	Status                 KubernetesClusterStatus   `json:"status"`
	// ...
}
```